### DISCUSSION TITLE
inheritance and a checkboxe

### DISCUSSION THREAD

**Discussion by jeagh** [alejandroautalan/pygubu-designer#261] [Q&A]
I have simple inheritance working where just buttons are involved but after a full day of trying to get a simple inherited checkbox app to work and failing it's now time to ask the experts.

This is my code that inherits two buttons and a checkbox. No problem with the buttons, the checkbox configure line generates this error:

   ValueError: ['onvalue', 'offvalue'] are not supported arguments. Look at the documentation for supported arguments.

The error has to be something very basic. I've searched the Internet for an answer an even pestered my AI friends but to no avail.  

`
#!/usr/bin/python3
from customtkinter_baseui import TestUI
import customtkinter

class Myapp(TestUI):
    def __init__(self):
        super().__init__()  # Initialise the parent class

        self.check_var = customtkinter.StringVar(value="on")        
        
        self.checkbox = self.builder.get_object("Ctkcheckbox1")
        self.checkbox.configure(variable=self.check_var, onvalue="on", offvalue="off")        

 `

**Comment by alejandroautalan** [comment]
Hello @jeagh, thanks for trying pygubu.

The problem here is that customtkinter restrict the use of some properties.
In Ctkcheckbox _onvalue_ and _offvalue_ are allowed at construction time:

https://github.com/TomSchimansky/CustomTkinter/blob/b9b57d95c7d3bb821a2260527315b6079161f5c9/customtkinter/windows/widgets/ctk_checkbox.py#L41-L42

But are not marked to be configurable using the _configure_ method:

https://github.com/TomSchimansky/CustomTkinter/blob/b9b57d95c7d3bb821a2260527315b6079161f5c9/customtkinter/windows/widgets/ctk_checkbox.py#L223

You can configure those properties using the designer, or ask the customtkinter project to fix the _configure_ method of Ctkcheckbox.

Regards,
Alejandro A

**Reply by jeagh** [reply]
Thank you alejandroautalan for your reply. I've spent another half a day trying to proceed with this. I also tried the switch widget but of course with the same problem.

Can you show me how to configure the checkbox properties in designer or direct me to a how-to site?

**Reply by alejandroautalan** [reply]
> Thank you alejandroautalan for your reply. I've spent another half a day trying to proceed with this. I also tried the switch widget but of course with the same problem.
> 
> Can you show me how to configure the checkbox properties in designer or direct me to a how-to site?

I made a simple example that you can study. It has buttons and checkboxes with Int and String variables.

![Captura desde 2024-06-25 23-20-51](https://github.com/alejandroautalan/pygubu-designer/assets/3482471/f438335f-f273-452b-bdd2-67ed7a864ad3)

[discussion-261.zip](https://github.com/user-attachments/files/15981581/discussion-261.zip)


If you are new with tkinter and pygubu-designer I recommend this [playlist](https://www.youtube.com/watch?v=1FuVPUIayZ8&list=PLd1leK7rsaC-MocOxaqDuiikpkM9I-YT9).
There is some documentation in the [wiki](https://github.com/alejandroautalan/pygubu-designer/wiki). Also there are many examples in [here](https://github.com/alejandroautalan/pygubu-designer/tree/master/examples).

If you have more questions, let me know.
Regards
Alejandro A.

**Reply by jeagh** [reply]
Thank you Alejandro. I'll study your example code and visit the links provided, I know about the wiki.

I've manged to get a working pygubu-designer checkbutton project working using a ttk checkbotton widget.
